eox_nelp/openedx_filters/xapi/filters.py

```diff
@@ -198,3 +199,45 @@ def run_filter(self, transformer, result):  # pylint: disable=arguments-differ,
         return {
             "result": result
         }
+
+
+class XApiContextFilter(PipelineStep):
+    """This filter is designed to update the context value for any event transformer.
+
+    How to set:
+        OPEN_EDX_FILTERS_CONFIG = {
+            "event_routing_backends.processors.xapi.transformer.xapi_transformer.get_context": {
+                "pipeline": ["eox_nelp.openedx_filters.xapi.filters.XApiContextFilter"],
+                "fail_silently": False,
+            },
+        }
+    """
+
+    def run_filter(self, transformer, result):  # pylint: disable=arguments-differ, unused-argument
+        """Modifies statement context for any event
+
+        Arguments:
+            transformer <XApiTransformer>: Transformer instance.
+            result <Context>: Context related to an event.
+
+        Returns:
+            Context: Modified context.
+        """
+        extra_context = getattr(settings, "XAPI_EXTRA_CONTEXT", {})
+
+        for key, value in extra_context.items():
+            if not hasattr(result, key):
+                break
+
+            attribute = getattr(result, key)
+
+            if isinstance(attribute, dict):
+                attribute.update(value)
+            else:
+                attribute = value
```

Comment:
If the attribute is not a dict it would override all the attribute so that setting has more priority. 
The filter would have a name like `SET`. Is like the operation you are realizing, but I am thinking it because if is dict it could preserve some other values of the dict...

Reply:
the run_filter name can not be changed  since openedx filter depends on that or if you refer to the class name this is just following the other filter name convention `XApiSomethingFilter ` so let me ask you for you what is an **openedx** filter ?

Reply:
I meant the class, but I am ok with the convention.

Reply:
keep in mind that is not a standard that is a convention that I adopted  since I think the `Filter` suffix makes reference to set, update or edit actions in the openedx-filter context, IMO filter was not the best name for that feature but it's what we have